media/base/android/java/src/org/chromium/media/MediaPlayerBridge.java

```diff
@@ -35,12 +35,18 @@
 */
 @JNINamespace("media")
 public class MediaPlayerBridge {
-
+    /**
+    * Give the host application a chance to take over MeidaPlayer.
+    */
     public static class ResourceLoadingFilter {
         public boolean shouldOverrideResourceLoading(
-                MediaPlayer mediaPlayer, Context context, Uri uri) {
+                ExternalMediaPlayer mediaPlayer, Context context, Uri uri) {
```

Comment:
We don't know which MediaPlayer is using, Maybe a flag need to be added, so it's more complicated.